File: api/AirSoft.Service/Contracts/Auth/SignUp/SignUpResponse.cs
using Store.Service.Contracts.Jwt.Model;
using Store.Service.Contracts.Models;

namespace Store.Service.Contracts.Auth.SignUp;

public class SignUpResponse
{
    public SignUpResponse(JwtResponse tokenData, UserData user, AuthProfileData profile)
    {
        TokenData = tokenData;
        User = user;
        Profile = profile;
    }

    public JwtResponse TokenData { get; }
    public UserData User { get; }
    public AuthProfileData Profile { get; }
}

File: spa/src/app/shared/services/dto-models/auth/sign-up/sign-up-response.ts
import { IProfileAuthData } from './../profile-auth-data';
import { ITokenData } from '../token-data';
import { IUserData } from '../user-data';

export interface ISignUpResponse {
  tokenData: ITokenData | null;
  user: IUserData | null;
  profile: IProfileAuthData | null;
}
